design/anoncreds.md

```diff
@@ -0,0 +1,507 @@
+    },
+...
+}
+```
+1. Lookup State Trie to get `schemaIssuerDid | SchemaMarker |schemaName | schemaVersion | schemaUUID` by `uuid`
+using Record2. 
+1. Lookup State Trie to get data by the key found above (Record1).
+
+So, we will have 2 lookups for each request. 
+
+### CRED_DEF
+
+The Definition of credentials for the given Schema by the given Issuer.
+It contains public keys (primary and revocation),
+signature type and reference to the Schema. 
+
+#### CRED_DEF txn
+```
+{
+    "data": {
+        "uuid":"TYzcdDLhCpGCYRHW82kjHd",
+        "publicKeys": {},
+        "schemaRef":"GEzcdDLhCpGCYRHW82kjHd",
+        "signatureType":"CL",
+        "oldKeyTrustTime": {                    (optional)
+            "from": "10", 
+            "to": "30",
+        }
+    },
+    
+    "reqMetadata": {
+        "issuerDid":"HHAD5g65TDQr1PPHHRoiGf",
+        .....
+    },
+    
+....
+}
+```
+* `uuid` is CredDef's UUID. It's different from `issuerDid`.
+* `oldKeyTrustTime` can be set each time the key is rotated and defines
+ `the intervals when we still can trust the previous value of the key`.
+ This is delta; all intervals are accumulated and appended in the State. 
+It is needed to deprecate credentials issued during the time when we suspect
+the keys were stolen.
+We can not always use revocation to deprecate old credentials, since revocation keys can
+be stolen as well.  
+ 
+
+#### Restrictions
+
+* Existing CredDef (identified by the CredDef `uuid`) can be modified/changed/evolved.
+That is rotation of keys is supported.
+* Only the `issuerDID` created the CredDef can modify it (that is we need to keep the ownership). 
+* It's not possible to create multiple entities with the same `uuid` (so, it's unique within the ledger).
+
+#### State
+
+We need to have two records for CredDef in State Trie in order to have
+1. Simple referencing of CredDef in the protocol (by CredDef DID)
+1. Requirements 8
+
+Record 1:
+* key: `credDefIssuerDid | CredDefMarker | schemaUUID | signatureType | credDefUUID` 
+* value: aggregated txn data plus `trustTime` as an array (each next `trustTme` is appended).
+
+Record 2:
+* key: `credDefUUID`
+* value: Record 1 key
+
+#### How oldKeyTrustTime works
+Let's assume that 
+* `key1` was issued at `timeA`
+* `key2` was issued at `timeC`, and we suspect that `key1` is stolen at `timeB`
+* `key3` was issued at `timeE`, and we suspect that `key2` is stolen at `timeD`
+
+So, we need to use (and return by `GET_CRED_DEF`) the following keys, depending on 
+the interval the `issuanceTime` belongs to:
+* [A,B] -> key1
+* [B,C] -> key2 
+* [C,D] -> key2
+* [D,E] -> key3
+* [E, current] -> key3
+
+So, the Credentials issued at intervals [B,C] and [D,E], that is at intervals
+when keys are suspicious, will not be verifiable anymore, because they were issued using key1 (key2),
+but the Verifies will use key2 (key3) for verification (as returned by `GET_CRED_DEF`). 
+
+The following txns will be put on Ledger:
```

Comment:
At what times is each of the transaction made like after A, B, etc?

Reply:
First one at timeA; second at timeC; third at timeE (will update it)